common/mrabbitmq/rabbitmq.go

```diff
@@ -0,0 +1,93 @@
+package mrabbitmq
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/LerianStudio/midaz/common/mlog"
+	"github.com/streadway/amqp"
+	"go.uber.org/zap"
+)
+
+// RabbitMQConnection is a hub which deal with rabbitmq connections.
+type RabbitMQConnection struct {
+	ConnectionStringSource string
+	Consumer               string
+	Producer               string
+	Channel                amqp.Channel
+	Connected              bool
+	Logger                 mlog.Logger
+}
+
+// Connect keeps a singleton connection with rabbitmq.
+func (rc *RabbitMQConnection) Connect(ctx context.Context) error {
+	rc.Logger.Info("Connecting on rabbitmq...")
+
+	conn, err := amqp.Dial(rc.ConnectionStringSource)
+	if err != nil {
+		rc.Logger.Fatal("failed to connect on rabbitmq", zap.Error(err))
+		return nil
+	}
+
+	defer conn.Close()
+
+	ch, err := conn.Channel()
+	if err != nil {
+		rc.Logger.Fatal("failed to open channel on rabbitmq", zap.Error(err))
+		return nil
+	}
+
+	defer ch.Close()
+
+	if ch == nil || !rc.healthCheck() {
+		rc.Connected = false
+		err := errors.New("can't connect rabbitmq")
+		rc.Logger.Fatalf("RabbitMQ.HealthCheck %v", zap.Error(err))
+
+		return err
+	}
+
+	rc.Logger.Info("Connected on rabbitmq ✅ \n")
+
+	rc.Connected = true
+
+	rc.Channel = *ch
```

Comment:
The problem here is that you are copying the value of an object that contains internal locks, which can cause concurrency problems. Instead of copying the value of the Channel, you could consider the pointer to `ch`, avoiding copying the locks.
```suggestion
	rc.Channel = ch
```